smelt.lic

```diff
@@ -26,79 +26,63 @@ class Smelt
 
     Flags.add('ingot-done', 'At last the metal appears to be thoroughly mixed')
 
-    get(@rod)
+    DRCC.get_crafting_item(@rod, @bag, @bag_items, @belt)
     if args.refine
-      get('flux')
-      type("pour my flux in crucible", "flux", true)
+      DRCC.get_crafting_item('flux', @bag, @bag_items, @belt)
+      work("pour my flux in crucible", "flux", true)
     else
-      type("stir crucible with my rod", "rod", false)
+      work("stir crucible with my rod", "rod", false)
     end
-    work
-    stow(@rod)
+    DRCC.stow_crafting_item(@rod, @bag, @belt)
   end
 
-  def get(name)
-    DRCC.get_crafting_item(name, @bag, @bag_items, @belt)
-  end
-
-  def stow(name)
-    DRCC.stow_crafting_item(name, @bag, @belt)
-  end
-
-  def type(command, item, stow)
-    @command = command
-    @item = item
-    @stow = stow
-  end
-
-  def work
+  def work(command, item, stow)
     loop do
       break if Flags['ingot-done']
-      case DRC.bput(@command, 'Pour what', 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
+      case DRC.bput(command, 'Pour what', 'You can only mix a crucible', 'clumps of molten metal', 'flickers and is unable to consume', 'needs more fuel', 'needs some more fuel', 'roundtime')
       when 'Pour what'
         DRC.message("Missing Flux")
         break
       when 'You can only mix a crucible'
         break
       when 'clumps of molten metal'
-        stow(@item) if @stow
-        type("turn crucible with my rod", "rod", false)
+        DRCC.stow_crafting_item(@item, @bag, @belt) if DRCI.in_left_hand?(@item)
+        @command = "turn crucible with my rod"
```

Comment:
Does this work? Line 42 is referencing `command`, not `@command`

Reply:
Yes, didnt work the other way.

Reply:
@Raykyn55 

Opening this back up if I could. Looking at the code, we only ever _assign_ instance variable `@command`. It is never actually used. Do a search of the code and double check. 

So if I follow the code - we enter the do loop. Our first `bput` uses the command `pour my flux in crucible`, which is passed in via method call and is a local variable. We then enter the case/when statements. Let's imagine it sees game output `clumps of molten metal`. It then assigns (for the first time), an instance variable called `@command` to the string `turn crucible with my rod`. We then come back to the top of the loop but our `bput` again is using the local `command` variable which is still set to `pour my flux in crucible`.

I thought maybe there was something funky with Ruby where a local variable is overridden by an instance variable of the same name but a test script proved that not to be true:

```ruby
class Test


  def initialize
    arg_definitions = [
      [

      ]
    ]

    # time_settings_pulls
    args = parse_args(arg_definitions)

    test("a command")
  end

  def test(command)
    2.times do
      echo command
      @command = "a new command"
    end
  end
end
```

```
>;drroom
--- Lich: drroom-test active.
[drroom-test: a command]
[drroom-test: a command]
--- Lich: drroom-test has exited.
```
This is how I believe it would flow just by looking at it. I can try to test it this evening if I can even figure out how smelting works!

Reply:
I don't know how to use methods like Robert suggested, and as you point out, what I tried didn't work. I went back to just defining instance variables.